src/JokeListClass.js

```diff
@@ -0,0 +1,124 @@
+import React, { Component } from "react";
+import axios from "axios";
+import _ from "lodash";
+import "./JokeList.css";
+import JokeClass from "./JokeClass";
+
+class JokeListClass extends Component {
+  state = {
+    numJokesToGet: 10,
+    jokes: [],
+  };
+
+  getJokes = async (stateJokes, numJokesToGet) => {
+    console.log("here");
+    let j = [...stateJokes];
+    let seenJokes = new Set();
+    try {
+      while (j.length < numJokesToGet) {
+        let res = await axios.get("https://icanhazdadjoke.com", {
+          headers: { Accept: "application/json" },
+        });
+        let { status, ...jokeObj } = res.data;
+
+        if (!seenJokes.has(jokeObj.id)) {
+          seenJokes.add(jokeObj.id);
+          j.push({ ...jokeObj, votes: 0 });
+        } else {
+          console.error("duplicate found!");
+        }
+      }
+      return j;
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  /* empty joke list and then call getJokes */
+  generateNewJokes = () => {
+    this.setState({ jokes: [] });
+  };
+
+  /* change vote for this id by delta (+1 or -1) */
+  vote = (id, delta) => {
+    console.log("vote updated for:", id);
+    this.setState((allJokes) =>
+      allJokes.jokes.map((j) =>
+        j.id === id ? { ...j, votes: j.votes + delta } : j
+      )
+    );
```

Comment:
this.setState({
      jokes: this.state.jokes.map(j => (
        j.id === id ? { ...j, votes: j.votes + delta } : j
      ))
    })